From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Thu, 11 May 2023 18:34:56 -0400
Subject: [PATCH] Fix binding compilation on Windows.

Need to %import <hamlib/config.h> so SWIG preprocessing of #if
statements matches source compilation.

Disable problematic "initializer element is not constant" #defines for
Lua bindings.
---
 bindings/hamlib.swg | 2 ++
 bindings/ignore.swg | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/bindings/hamlib.swg b/bindings/hamlib.swg
index 05284c28..b9df403f 100644
--- a/bindings/hamlib.swg
+++ b/bindings/hamlib.swg
@@ -220,6 +220,8 @@
 %immutable rot_caps::version;
 %immutable rot_caps::copyright;
 
+%import <hamlib/config.h>
+
 %include <hamlib/rig_dll.h>
 
 %include <hamlib/riglist.h>
diff --git a/bindings/ignore.swg b/bindings/ignore.swg
index 37d6c96d..330a44bb 100644
--- a/bindings/ignore.swg
+++ b/bindings/ignore.swg
@@ -219,5 +219,9 @@
 %ignore Rig::set_level(char const *,int,vfo_t);
 %ignore Rig::set_parm(setting_t,int);
 %ignore Rig::set_parm(char const *,int);
+%ignore RIG_LEVEL_FLOAT_LIST;
+%ignore RIG_LEVEL_READONLY_LIST;
+%ignore RIG_LEVEL_IS_FLOAT;
+%ignore RIG_LEVEL_SET;
 #endif
 
-- 
2.42.0

